player/configfiles.c

```diff
@@ -313,7 +320,7 @@ void mp_write_watch_later_conf(struct MPContext *mpctx)
         goto exit;
     }
 
-    mp_mk_config_dir(mpctx->global, MP_WATCH_LATER_CONF);
+    mp_mk_config_dir(mpctx->global, mpctx->cached_watch_later_configdir);
```

Comment:
I think cached_watch_later_configdir is not necessarily set yet here, depending on whether mp_get_playback_resume_config_filename was called before independently.